### PR TITLE
Stepper: Replace domain upsell exitFlow() usage with stepper submit() function

### PR DESCRIPTION
**Pull request by agrullon95**
Related to #

## Proposed Changes

* The changes in this PR remove the `exitFlow` function from the domain upsell flow and replaces it with the `submit` function. We should only navigate between screens/pages in the Stepper framework using the `submit` function.

## Testing Instructions

* Checkout this branch.
* Run `yarn start`.
* Create a new "free flow" launchpad enabled site or use an existing free flow site that hasn't been launched yet.
* Click the "Choose a domain" checklist task on the launchpad screen.
* Click the "Back" button on the top left side of the domains selection screen and verify it navigates back to the launchpad screen.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by alshakero** [comment]
Beautiful ✨

**Comment by agrullon95** [comment]
@alshakero - We can follow up with a PR to remove exitFlow() from `DomainsStep` and modify the affected flows.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx [hunk 1]
```diff
@@ -245,7 +245,9 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 		}
 
 		if ( flow === DOMAIN_UPSELL_FLOW ) {
-			return exitFlow?.( `/setup/${ flowToReturnTo }/launchpad?siteSlug=${ siteSlug }` );
+			return submit?.( {
+				exitFlowTo: `/setup/${ flowToReturnTo }/launchpad?siteSlug=${ siteSlug }`,
+			} );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx]: This doesn't add much value, the step is still making decisions based on the flow. And exiting the flow. 

[Now that `goBack` isn't deprecated anymore](https://github.com/Automattic/wp-calypso/pull/77072), feel free to just call it. And implement it in the `DOMAIN_UPSELL_FLOW` flow. 

LMK if this is unclear, we can do a quick huddle.

**Review by alshakero — CHANGES_REQUESTED** [review]

File: client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx (hunk 1 above)
Comment (agrullon95) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx]: @alshakero - Makes sense. I'll update this to use goBack instead and move the launchpad url into the domain_upsell flow file.

**Review by alshakero — APPROVED** [review]
Let's ship this. It still exits the flow. But to fix this, we'd need to change a few flows.

**Pull request merged**